base/memory/memory_pressure_monitor_endless.cc

```diff
@@ -126,8 +135,9 @@ int MemoryPressureMonitor::GetUsedMemoryInPercent() {
   // unless it is dirty or it's a minimal portion which is required.
   file_memory -= info.dirty + kMinFileMemory;
 
-  // Available memory is the sum of free and easy reclaimable memory (no swap).
-  int available_memory = info.free + file_memory;
+  // Available memory is the sum of free, swap and easy reclaimable memory.
+  int available_memory = info.free
+    + std::min(info.free * kSwapWeight, info.swap_free) * (kSwapWeight - 1) / kSwapWeight;
```

Comment:
Here I would augment our idea of free memory (in both places) with the file_memory number, for the same reasons that Chromium states above. Non dirty cached pages can be evicted under memory pressure. If we don't consider them free we will underestimate the available space significantly.

Reply:
Sorry, that was a sloppy mistake on my patch, I never meant to remove `file_memory`

Reply:
Fixed